e2e/test/iothub/command/CommandE2ETests.cs

```diff
@@ -0,0 +1,323 @@
+        }
+
+
+        [LoggedTestMethod]
+        public async Task Command_DeviceUnsubscribes_MqttWs()
+        {
+            await SendCommandAndUnsubscribeAsync(Client.TransportType.Mqtt_WebSocket_Only, SubscribeAndUnsubscribeCommandAsync).ConfigureAwait(false);
+        }
+
+        [LoggedTestMethod]
+        public async Task Command_DeviceReceivesCommandAndResponseWithComponent_Mqtt()
+        {
+            await SendCommandAndRespondAsync(Client.TransportType.Mqtt_Tcp_Only, SetDeviceReceiveCommandWithComponentAsync, withComponent: true).ConfigureAwait(false);
+        }
+
+        [LoggedTestMethod]
+        public async Task Command_DeviceReceivesCommandAndResponseWithComponent_MqttWs()
+        {
+            await SendCommandAndRespondAsync(Client.TransportType.Mqtt_WebSocket_Only, SetDeviceReceiveCommandWithComponentAsync, withComponent: true).ConfigureAwait(false);
+        }
+
+
+        [LoggedTestMethod]
+        public async Task Command_DeviceUnsubscribesWithComponent_Mqtt()
+        {
+            await SendCommandAndUnsubscribeAsync(Client.TransportType.Mqtt_Tcp_Only, SubscribeAndUnsubscribeCommandWithComponentAsync, withComponent: true).ConfigureAwait(false);
+        }
+
+
+        [LoggedTestMethod]
+        public async Task Command_DeviceUnsubscribesWithComponent_MqttWs()
+        {
+            await SendCommandAndUnsubscribeAsync(Client.TransportType.Mqtt_WebSocket_Only, SubscribeAndUnsubscribeCommandWithComponentAsync, withComponent: true).ConfigureAwait(false);
+        }
+
+        public static async Task ServiceSendCommandAndVerifyResponseAsync(string deviceId, string methodName, MsTestLogger logger, TimeSpan responseTimeout = default, ServiceClientTransportSettings serviceClientTransportSettings = default)
+        {
+            ServiceClient serviceClient = null;
+            if (serviceClientTransportSettings == default)
+            {
+                serviceClient = ServiceClient.CreateFromConnectionString(Configuration.IoTHub.ConnectionString);
+            }
+            else
+            {
+                serviceClient = ServiceClient.CreateFromConnectionString(Configuration.IoTHub.ConnectionString, TransportType.Amqp, serviceClientTransportSettings);
+            }
+
+            TimeSpan methodTimeout = responseTimeout == default ? s_defaultCommandTimeoutMinutes : responseTimeout;
+            logger.Trace($"{nameof(ServiceSendCommandAndVerifyResponseAsync)}: Invoke method {methodName}.");
+            try
+            {
+                CloudToDeviceMethodResult response =
+                    await serviceClient.InvokeDeviceMethodAsync(
+                        deviceId,
+                        new CloudToDeviceMethod(methodName, methodTimeout).SetPayloadJson(null)).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                if (!(ex is DeviceNotFoundException))
```

Comment:
why are we expecting a `DeviceNotFoundException` here?

Reply:
Copied over from Methods test. Removed